contracts/chain-adapters/Solana_Adapter.sol

```diff
@@ -0,0 +1,193 @@
+ * called via delegatecall, which will execute this contract's logic within the context of the originating contract.
+ * For example, the HubPool will delegatecall these functions, therefore it's only necessary that the HubPool's methods
+ * that call this contract's logic guard against reentrancy.
+ * @custom:security-contact bugs@across.to
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Solana_Adapter is AdapterInterface, CircleCCTPAdapter {
+    /**
+     * @notice The official Circle CCTP MessageTransmitter contract endpoint.
+     * @dev Posted officially here: https://developers.circle.com/stablecoins/docs/evm-smart-contracts
+     */
+    // solhint-disable-next-line immutable-vars-naming
+    IMessageTransmitter public immutable cctpMessageTransmitter;
+
+    // Solana spoke pool address, decoded from Base58 to bytes32.
+    bytes32 public immutable SOLANA_SPOKE_POOL_BYTES32;
+
+    // Solana spoke pool address, mapped to its EVM address representation.
+    address public immutable SOLANA_SPOKE_POOL_ADDRESS;
+
+    // USDC mint address on Solana, decoded from Base58 to bytes32.
+    bytes32 public immutable SOLANA_USDC_BYTES32;
+
+    // USDC mint address on Solana, mapped to its EVM address representation.
+    address public immutable SOLANA_USDC_ADDRESS;
+
+    // USDC token address on Solana for the spoke pool (vault ATA), decoded from Base58 to bytes32.
+    bytes32 public immutable SOLANA_SPOKE_POOL_USDC_VAULT;
+
+    // Custom errors for constructor argument validation.
+    error InvalidCctpTokenMessenger(address tokenMessenger);
+    error InvalidCctpMessageTransmitter(address messageTransmitter);
+
+    // Custom errors for relayMessage validation.
+    error InvalidRelayMessageTarget(address target);
+    error InvalidOriginToken(address originToken);
+    error InvalidDestinationChainId(uint256 destinationChainId);
+
+    // Custom errors for relayTokens validation.
+    error InvalidL1Token(address l1Token);
+    error InvalidL2Token(address l2Token);
+    error InvalidAmount(uint256 amount);
+    error InvalidTokenRecipient(address to);
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1Usdc USDC address on L1.
+     * @param _cctpTokenMessenger TokenMessenger contract to bridge tokens via CCTP.
+     * @param _cctpMessageTransmitter MessageTransmitter contract to bridge messages via CCTP.
+     * @param solanaSpokePool Solana spoke pool address, decoded from Base58 to bytes32.
+     * @param solanaUsdc USDC mint address on Solana, decoded from Base58 to bytes32.
+     * @param solanaSpokePoolUsdcVault USDC token address on Solana for the spoke pool, decoded from Base58 to bytes32.
+     */
+    constructor(
+        IERC20 _l1Usdc,
+        ITokenMessenger _cctpTokenMessenger,
+        IMessageTransmitter _cctpMessageTransmitter,
+        bytes32 solanaSpokePool,
+        bytes32 solanaUsdc,
+        bytes32 solanaSpokePoolUsdcVault
```

Comment:
nit: it might be easier/cleaner/simpler/more consistant to just have these hard coded as constants, as done in some of the other [adapters](https://github.com/across-protocol/contracts/blob/5f08c27dd09f8c674ccc4af39d8b6b7c87a0444e/contracts/chain-adapters/Optimism_Adapter.sol#L52).

Reply:
I left them as immutables so that its easier to test integration flow on testnets that would have different USDC and we don't have a proper authority for the spoke yet.